core/src/main/java/tc/oc/pgm/api/channels/Channel.java

```diff
@@ -0,0 +1,118 @@
+package tc.oc.pgm.api.channels;
+
+import static net.kyori.adventure.text.Component.text;
+
+import cloud.commandframework.arguments.standard.StringArgument;
+import cloud.commandframework.paper.PaperCommandManager;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.event.ChannelMessageEvent;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.util.Players;
+import tc.oc.pgm.util.named.NameStyle;
+
+public interface Channel<T> {
```

Comment:
This could extend Adventure's `Audience`.

Reply:
given the extra overhead that a channel implies, it probably shouldn't.
You don't want someone treating this like it was just another casual audience, when it may not act like one at all (eg: requirement for a *target* to send within the channel). Effectively speaking the only channel that could properly implement audience would be global chat